### PR TITLE
RUM-5553: Create Benchmark module to collect performance metrics

### PR DESCRIPTION
**Pull request by ambushwork**
### What does this PR do?

Creates a module in SDK for the metrics benchmarks

### Motivation

[RUM-5553]

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

[RUM-5553]: https://datadoghq.atlassian.net/browse/RUM-5553?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -53,3 +53,4 @@ include(":tools:unit")
 include(":tools:noopfactory")
 include(":tools:javabackport")
 include(":tools:lint")
+include(":benchmark")
```
Comment (xgouchet) [inline settings.gradle.kts:56]: > [!NOTE]
> Let's make it a `:tools:benchmark`  module instead to limit the number of top level folders in the repo

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: settings.gradle.kts (hunk 1 above)
Comment (ambushwork) [reply settings.gradle.kts]: updated

**Review by mariusc83 — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

File: tools/benchmark/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        java17()
+    }
+}
+
+dependencies {
+
+    implementation(libs.kotlin)
+    implementation(libs.okHttp)
+    api(libs.openTelemetrySdk)
+    api(libs.openTelemetryApi)
```
Comment (0xnm) [inline tools/benchmark/build.gradle.kts:35]: this probably should be `implementation`, unless we want to re-export definitions in some module which will consume this one

File: tools/benchmark/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    implementation(libs.gson)
+    compileOnly(libs.detektApi)
```
Comment (0xnm) [inline tools/benchmark/build.gradle.kts:37]: is it needed?

File: tools/benchmark/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            attribute(
+                com.android.build.api.attributes.ProductFlavorAttr.of("platform"),
+                objects.named("jvm")
+            )
+        }
+    }
+
+    testImplementation(libs.bundles.jUnit5)
+    testImplementation(libs.bundles.testTools)
+    testImplementation(libs.detektTest)
```
Comment (0xnm) [inline tools/benchmark/build.gradle.kts:49]: and this is not needed as well, I guess.

File: tools/benchmark/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    testImplementation(libs.robolectric)
```
Comment (0xnm) [inline tools/benchmark/build.gradle.kts:50]: probably also not needed?

**Pull request merged**